### PR TITLE
Added support for arrow keys in class listing

### LINKED ISSUE #66: add key binding to class listing

### PR DESCRIPTION
**Pull request by ilbinek**
Changing selected class with arrow keys now actually triggers update in the buffer views.

Tried other methods too, but this one triggers the update only when the arrow key us released, so the update is called only once.
Trying to do this with other listeners/methods resulted in a lot of scheduled updates and lag.

Fixes #66

### CODE REVIEW

File: runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java [hunk 1]
```diff
@@ -17,12 +17,7 @@
 import javax.swing.event.DocumentEvent;
 import javax.swing.event.DocumentListener;
 import java.awt.*;
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
-import java.awt.event.ComponentAdapter;
-import java.awt.event.ComponentEvent;
-import java.awt.event.MouseAdapter;
-import java.awt.event.MouseEvent;
+import java.awt.event.*;
```
Comment (judovana) [inline runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java:20]: Please don't. Adjust your idea so it do not compact imports and keep enumerated listing.
In idea it is hidden, there is some maximal occurence of imports before the comapcting is used, and that needs to be set to  some 9999

File: runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java [hunk 2]
```diff
@@ -143,6 +138,29 @@ protected Void doInBackground() throws Exception {
...
+        filteredClassesJlist.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyReleased(KeyEvent e) {
+                if (e.getKeyCode() == KeyEvent.VK_UP || e.getKeyCode() == KeyEvent.VK_DOWN) {
+                    final String name = filteredClassesJlist.getSelectedValue();
+                    if (name != null || filteredClassesJlist.getSelectedIndex() != -1) {
+                        new SwingWorker<Void, Void>() {
+                            @Override
+                            protected Void doInBackground() throws Exception {
+                                try {
```
Comment (judovana) [inline runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java:150]: Can not be reused the original worker which is already done for mouse listener?
Also... what about pgup/pgdown? 
Also, how does it behave when you are already on top/bottom of the list? I guess the item reload anyway. (but this bug may be hidden already in original mouselistener)
How does it behaves on single class programs? Maybe click/space/enter can enforce refresh?

Comment (ilbinek) [reply runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java:150]: I can extract the worker into it's own function and call it from both mouse listener and key listener.

Pgdown/pgup does not trigger update, will fix.

When on top/bottom, it does trigger an update, same as clicking the same class again with mouse. Does it need fixing?

On a single class program, clicking the class/pgup/pgdown/arrow up/down would refresh it. Can implement enter as a trigger for update.

Comment (judovana) [reply runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/BytecodeDecompilerView.java:150]: Please extract the worker.
pgu/dow - ty!
the top/bottom/single  behaviour sounds good.

**Pull request merged**